### ISSUE TITLE
oce should use build-in .leap.seconds

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1310]
In `R/oce.R` near line 2676, there is a definition of the leap seconds that have occurred after 1981.  This is fragile code, because it relies on developers checking for leap seconds every year or so. But there's no need, because R has `.leap.seconds`, which holds the same data (going back further).

The code in `R/oce.R` now reads
```R
leaps <- as.POSIXct(strptime(c("1981-07-01", "1982-07-01", "1983-07-01", "1985-07-01", 
   "1987-12-31", "1989-12-31", "1990-12-31", "1992-07-01", "1993-07-01", "1994-07-01",
    "1995-12-31", "1997-07-01", "1998-12-31", "2005-12-31", "2008-12-31", "2012-07-01",
    "2015-07-01", "2016-12-31"), format="%Y-%m-%d", tz="UTC"))
```
and R defines
```R
> .leap.seconds
 [1] "1972-06-30 21:00:00 ADT" "1972-12-31 20:00:00 AST"
 [3] "1973-12-31 20:00:00 AST" "1974-12-31 20:00:00 AST"
 [5] "1975-12-31 20:00:00 AST" "1976-12-31 20:00:00 AST"
 [7] "1977-12-31 20:00:00 AST" "1978-12-31 20:00:00 AST"
 [9] "1979-12-31 20:00:00 AST" "1981-06-30 21:00:00 ADT"
[11] "1982-06-30 21:00:00 ADT" "1983-06-30 21:00:00 ADT"
[13] "1985-06-30 21:00:00 ADT" "1987-12-31 20:00:00 AST"
[15] "1989-12-31 20:00:00 AST" "1990-12-31 20:00:00 AST"
[17] "1992-06-30 21:00:00 ADT" "1993-06-30 21:00:00 ADT"
[19] "1994-06-30 21:00:00 ADT" "1995-12-31 20:00:00 AST"
[21] "1997-06-30 21:00:00 ADT" "1998-12-31 20:00:00 AST"
[23] "2005-12-31 20:00:00 AST" "2008-12-31 20:00:00 AST"
[25] "2012-06-30 21:00:00 ADT" "2015-06-30 21:00:00 ADT"
[27] "2016-12-31 20:00:00 AST"
```

**Comment by dankelley** [comment]
I'll do this today. I want to find a reference value to use in the build-time tests, and when I do that, I'll put comments in the test file and also in the docs.

**Comment by dankelley** [comment]
I'm looking now at te site https://www.labsat.co.uk/index.php/en/gps-time-calculator but I'd prefer to find something from a government body. Anyway, at this site I get as shown (which does not match my test value in `tests/testthat/test_misc.R` line 36.

<img width="901" alt="screen shot 2017-10-14 at 9 40 29 am" src="https://user-images.githubusercontent.com/99469/31575750-d9d26ec8-b0c3-11e7-8e50-851412b007f4.png">

**Comment by dankelley** [comment]
Fixed in "develop" branch, commit 42718505c806620aeaa20b08486b6dc31b9edfc1

I have not found what I consider to be an authoritative and simple website to quote in the docs, but in the test suite I have some specifics that will be enough to help if problems arise in the future; a snap is below:

<img width="909" alt="screen shot 2017-10-14 at 10 47 38 am" src="https://user-images.githubusercontent.com/99469/31576279-a87fe738-b0cd-11e7-88ce-8a810280c519.png">
